src/analysis/retail/priest/holy/modules/talents/Oracle/PremonitionOfInsight.tsx

```diff
@@ -43,21 +43,30 @@ class PremonitionOfInsight extends Analyzer {
     this.addEventListener(Events.cast.by(SELECTED_PLAYER), this.handleOnCast);
   }
 
-  //this function is really messy because i wrote it at 2 am
-  //while dealing with an undefined error for 3 hours
   handleOnCast(event: CastEvent) {
     if (!removesInsightCharge(event)) {
       return;
     }
+
     const spellId = event.ability.guid;
     const effCDR = this.spellUsable.reduceCooldown(spellId, this.scaledInsightCDR);
 
     this.insightReducBySpell[spellId] = this.insightReducBySpell[spellId] || 0;
 
     this.insightStackNumber += 1;
 
-    const stackNumx = this.insightStackNumber;
-    const castSpellIdx = spellId;
+    /*have to hard code hf because holy fire's cast time causes it to
+      bug geteffectiveCDR */
+    if (spellId === SPELLS.HOLY_FIRE.id) {
```

Comment:
that sounds like a fun bug. how does the holy fire cdr work in game?

Reply:
It is just a flat 10s CD, Insight with Fatebender is a 9.8s reduction on cast and while the ability is successfully reset getEffectiveCDR never sees any CDR amount, either because it polls before the cast is finish or after the CDR is already applied and fully reset the ability. I'm not sure if any other casted abilities are bugged, I will have to get logs and check more abilities at some point in the season.

![image](https://github.com/user-attachments/assets/e820bba4-f2e4-4a98-a5af-ec4fccbf9f7b)

Reply:
fwiw i think this is broken because the cooldown of holy fire is not implemented: https://github.com/WoWAnalyzer/WoWAnalyzer/blob/1d28aa66bb726928ad213505b57c1f37624a4e25/src/analysis/retail/priest/holy/modules/Abilities.ts#L216-L226

but that can be a later thing

Reply:
Ah, I will have to go through these spells this week. I thought it was something that was auto pulled from somewhere.